packages/gestalt-datepicker/src/InternalDateField.js

```diff
@@ -0,0 +1,285 @@
+  name,
+  onBlur,
+  onChange,
+  onClearInput,
+  onError,
+  onFocus,
+  readOnly = false,
+  size = 'md',
+  value,
+}: InternalDateFieldProps): Node {
+  const [focused, setFocused] = useState(false);
+
+  return (
+    <Box>
+      <label
+        className={classnames(styles.label, {
+          [styles.visuallyHidden]: labelDisplay === 'hidden',
+        })}
+        htmlFor={id}
+      >
+        <div className={styles.formLabel}>
+          <Text size="100">{label}</Text>
+        </div>
+      </label>
+      <Box position="relative" display="flex" alignItems="center">
+        <MUIDatePicker
+          onChange={onChange}
+          disabled={disabled}
+          readOnly={readOnly}
+          onError={(error) => onError?.({ errorMessage: error, value })}
+          errorMessage={!!errorMessage}
+          maxDate={maxDate}
+          minDate={minDate}
+          disableFuture={disableRange === 'disableFuture'}
+          disablePast={disableRange === 'disablePast'}
+          slots={{ field: CustomDateField }}
+          value={value}
+          passthroughProps={{
+            autoComplete,
+            size,
+            id,
+            errorMessage: !!errorMessage,
+            enterKeyHint: mobileEnterKeyHint,
+            name,
+            onBlur,
+            onFocus,
+          }}
+          viewRenderers={null}
+        />
+        {!disabled && !readOnly ? (
+          <div className={classnames(styles.actionButtonContainer)}>
+            <Box alignItems="center" display="flex" height="100%" marginEnd={2} rounding="circle">
+              <TapArea
+                accessibilityLabel="Clear date"
+                onBlur={() => setFocused(false)}
+                onFocus={() => setFocused(true)}
+                onKeyDown={({ event }) => {
+                  if ([ENTER, SPACE].includes(event.keyCode)) onClearInput();
+                  if (event.keyCode !== TAB) event.preventDefault();
+                }}
+                onMouseEnter={() => setFocused(true)}
+                onMouseLeave={() => setFocused(false)}
+                onTap={() => onClearInput()}
+                rounding="circle"
+                tapStyle="compress"
+              >
+                <Pog
+                  accessibilityLabel=""
+                  bgColor={focused ? 'lightGray' : 'transparent'}
+                  icon="cancel"
+                  iconColor="darkGray"
+                  size="xs"
+                />
+              </TapArea>
+            </Box>
+          </div>
+        ) : null}
+      </Box>
+      {helperText && !errorMessage ? (
+        <Box marginTop={2} id={`${id}-helperText`}>
+          <Flex gap={4}>
+            <Flex.Item flex="grow">
+              {helperText ? (
+                <Text color="subtle" size="100">
+                  {helperText}
+                </Text>
+              ) : null}
+            </Flex.Item>
+          </Flex>
+        </Box>
+      ) : null}
+      {errorMessage ? (
```

Comment:
Is this our existing input code and styling? Maybe we can have a `FieldWrapper` and the only child is the content? We can probably reusue in our other form inputs as well

Reply:
The gestalt package is not share with gestalt-datepicker so I duplicated all code. There's no really point of drying unless readability, which is also a good reason. If we move forward, I'll be a follow up.